From: =?utf-8?b?0JvQsNCx0L7RgNCw0YLQvtGA0LjRjyA1MA==?= <team@lab50.net>
Date: Sat, 26 Dec 2020 11:13:50 +0300
Subject: Dropped deprecated xbuild

---
Index: mono/runtime/Makefile.am
===================================================================
--- mono.orig/runtime/Makefile.am
+++ mono/runtime/Makefile.am
@@ -34,7 +34,7 @@ SUPPORT_FILES = $(symlinks) mono-wrapper
 build_profiles = 
 
 if INSTALL_4_x
-build_profiles += net_4_x xbuild_12 xbuild_14
+build_profiles += net_4_x
 net_profile = net_4_x
 endif
 
Index: mono/mcs/Makefile
===================================================================
--- mono.orig/mcs/Makefile
+++ mono/mcs/Makefile
@@ -44,7 +44,7 @@ dir-check:
 
 # fun specialty targets
 
-PROFILES = net_4_x xbuild_12 xbuild_14
+PROFILES = net_4_x
 
 .PHONY: all-profiles compiler-test-profiles run-compiler-test-profiles $(STD_TARGETS:=-profiles)
 all-profiles compiler-test-profiles run-compiler-test-profiles $(STD_TARGETS:=-profiles): %-profiles: profiles-do--%
Index: mono/mcs/tools/xbuild/Makefile
===================================================================
--- mono.orig/mcs/tools/xbuild/Makefile
+++ mono/mcs/tools/xbuild/Makefile
@@ -3,17 +3,14 @@ SUBDIRS =
 include ../../build/rules.make
 NO_TESTS = yes
 
+XBUILD_VERSION = 14.0
+FRAMEWORK_VERSION = 4.5
 include xbuild.make
 
 LOCAL_MCS_FLAGS =
 LIB_REFS = $(XBUILD_FRAMEWORK) $(XBUILD_UTILITIES) $(XBUILD_ENGINE) $(XBUILD_MSTASKS) $(PARENT_PROFILE)System $(PARENT_PROFILE)System.Core
 PROGRAM = xbuild.exe
 
-include ../../build/executable.make
-
-XBUILD_DIR=.
-include $(XBUILD_DIR)/xbuild_test.make
-
 install-local: install-extras
 
 PORTABLE_TARGETS_SRC:=data/Portable/Targets
@@ -34,10 +31,10 @@ install-extras: install-versioned-files
 endif
 
 #install files into xbuild's versioned locations
-install-versioned-files: install-bin-data install-nuget-imports
+install-versioned-files: install-nuget-imports
 
 #install files that are only installed once across all xbuild versions
-install-global-files: install-frameworks install-web-targets install-pcl-targets install-pcl5-framework install-nuget-targets install-msbuild-specific-files
+install-global-files: install-web-targets install-pcl-targets install-pcl5-framework install-nuget-targets install-msbuild-specific-files
 
 install-bin-data:
 	$(MKINSTALLDIRS) $(DESTDIR)$(XBUILD_BIN_DIR)/MSBuild
@@ -203,3 +200,6 @@ EXTRA_DISTFILES = \
 	$(NUGET_BUILDTASKS_REPO_DIR)/src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.props \
 	xbuild.make \
 	xbuild_test.make
+
+.PHONY: install-global-files install-web-targets install-pcl-targets install-pcl5-framework install-nuget-targets install-msbuild-specific-files install-versioned-files install-nuget-imports install-extras
+
