From: Debian Mono Group <pkg-mono-group@lists.alioth.debian.org>
Date: Sun, 13 Dec 2020 11:39:12 +0300
Subject: fix-tests

===================================================================
---
 mono/unit-tests/Makefile.am | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mono/unit-tests/Makefile.am b/mono/unit-tests/Makefile.am
index 85fff69..a21af8b 100644
--- a/mono/unit-tests/Makefile.am
+++ b/mono/unit-tests/Makefile.am
@@ -95,18 +95,18 @@ test_mono_handle_LDFLAGS = $(test_ldflags)
 
 test_mono_callspec_SOURCES = main.c
 test_mono_callspec_CFLAGS = $(AM_CFLAGS) -DMAIN=test_mono_callspec_main
-test_mono_callspec_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-callspec.lo
+test_mono_callspec_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-callspec.lo -lstdc++
 test_mono_callspec_LDFLAGS = $(test_ldflags)
 test_mono_callspec_DEPENDENCIES = callspec.exe
 
 test_mono_string_SOURCES = main.c
 test_mono_string_CFLAGS = $(test_cflags) -DMAIN=test_mono_string_main
-test_mono_string_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-string.lo
+test_mono_string_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-string.lo -lstdc++
 test_mono_string_LDFLAGS = $(test_ldflags)
 
 test_mono_embed_SOURCES = main.c
 test_mono_embed_CFLAGS = $(test_cflags) -DMAIN=test_mono_embed_main
-test_mono_embed_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-embed.lo
+test_mono_embed_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-mono-embed.lo -lstdc++
 test_mono_embed_LDFLAGS = $(test_ldflags)
 test_mono_embed_DEPENDENCIES = test-embed.exe
 
@@ -117,7 +117,7 @@ test_path_LDFLAGS = $(test_ldflags)
 
 test_embed_invoke_SOURCES = main.c
 test_embed_invoke_CFLAGS = $(test_cflags) -DMAIN=test_mono_embed_invoke_main
-test_embed_invoke_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-embed-invoke.lo
+test_embed_invoke_LDADD = $(test_ldadd) $(mini_libs) $(sgen_libs) libtestlib_la-test-embed-invoke.lo -lstdc++
 test_embed_invoke_LDFLAGS = $(test_ldflags)
 test_embed_invoke_DEPENDENCIES = test-embed-invoke-cs.exe
 
